spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -236,6 +263,18 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
   def parameters(): (Array[Tensor[T]], Array[Tensor[T]]) = null
 
   /**
+   * This function returns the array of trainable parameters.
+   * @return trainable parameters
+   */
+  def trainableParameters(): (Array[Tensor[T]], Array[Tensor[T]]) = {
```

Comment:
parameters() is override in those layers with weight or bias, if I use optional flag, I need to modify many files in order to filter trainable. Instead, trainableParameters is only needed in AbstractModule.scala and Container.scala